sycl/include/CL/sycl/stream.hpp

```diff
@@ -72,16 +72,26 @@ using GlobalOffsetAccessorT =
              cl::sycl::access::target::global_buffer,
              cl::sycl::access::placeholder::false_t>;
 
+// Read first 2 bytes of flush buffer to get buffer offset.
+// TODO: Should be optimized to the following:
+//   return *reinterpret_cast<uint16_t *>(&GlobalFlushBuf[WIOffset]);
+// when an issue with device code compilation using this optimization is fixed.
 inline unsigned GetFlushBufOffset(const GlobalBufAccessorT &GlobalFlushBuf,
                                   unsigned WIOffset) {
-  return (((uint8_t)GlobalFlushBuf[WIOffset]) << 8) +
-         (uint8_t)GlobalFlushBuf[WIOffset + 1];
+  return ((static_cast<unsigned>(static_cast<uint8_t>(GlobalFlushBuf[WIOffset]))
```

Comment:
```suggestion
  return ((static_cast<unsigned>(GlobalFlushBuf[WIOffset])
```
Can't this be directly cast to unsigned?

Reply:
No, unfortunately. Here I found an explanation: https://stackoverflow.com/a/21769421.
So, negative char is casted to negative integer first and after that to unsigned integer. If we save 255 into char then we will get `2^32 - 1` in this case.